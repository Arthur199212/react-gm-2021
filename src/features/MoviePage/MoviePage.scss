@import 'variables';

.app-movie-page {
  .main {
    align-items: center;
    display: flex;
    flex-direction: column;

    .search-box-container {
      margin-bottom: 2rem;

      .search-header {
        width: 100%;

        .title {
          font-size: 2rem;
          font-weight: $bold;
          margin: 4rem auto 2rem;
          text-align: center;
        }
      }
    }
  }
}
