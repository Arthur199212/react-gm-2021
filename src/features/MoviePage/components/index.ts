export * from './Header'
export * from './MovieDetails'
