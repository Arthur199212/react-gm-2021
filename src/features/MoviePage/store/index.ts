export * from './moviePage.selectors'
export * from './moviePage.slice'
export * from './moviePage.thunks'
