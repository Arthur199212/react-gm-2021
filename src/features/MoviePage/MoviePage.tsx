import { useRouter } from 'next/router'
import React, { useEffect } from 'react'
import { Filters, Footer, SearchResults } from '@app/components'
import { useAppDispatch, useAppSelector } from '@app/hooks'
import { Header, MovieDetails } from './components'
import { fetchMovieThunk, selectMovie } from './store'

export const MoviePage = () => {
  const dispatch = useAppDispatch()
  const {
    query: { movieId }
  } = useRouter()
  const movie = useAppSelector(selectMovie)

  useEffect(() => {
    window.scrollTo(0, 0)

    if (movieId && movieId !== String(movie?.id)) {
      dispatch(fetchMovieThunk(movieId as string))
    }
  }, [dispatch, movie, movieId])

  return (
    <>
      <div className='app-movie-page content'>
        <Header />
        <main className='main'>
          <MovieDetails />
          <div className='container'>
            <Filters />
            <SearchResults />
          </div>
        </main>
      </div>
      <Footer />
    </>
  )
}
