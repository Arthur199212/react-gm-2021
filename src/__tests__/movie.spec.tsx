import React from 'react'
import MoviePage, { getServerSideProps } from '@app/pages/movie/[movieId]'
import { render } from '@app/tests/testing-utils'
import { MOCK_MOVIE } from '@app/tests/mocks/mock-data'

describe('Movie Page', () => {
  it('should render properly', async () => {
    const { asFragment } = render(<MoviePage />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('getServerSideProps should work properly', async () => {
    const {
      props: { preloadedState }
    } = await getServerSideProps({ query: { movieId: '1' } } as any)

    expect(preloadedState.movie).toEqual({ movie: MOCK_MOVIE, status: 'success' })
  })
})
