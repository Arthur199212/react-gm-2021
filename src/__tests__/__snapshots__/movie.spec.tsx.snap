// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Page should render properly 1`] = `
<DocumentFragment>
  <div
    class="app-movie-page content"
  >
    <header
      class="header"
    >
      <div
        class="header-container container"
      >
        <span
          class="title"
        >
          MovieSearchApp
        </span>
        <i
          class="search-icon fas fa-search"
        />
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="movie-details-container container"
      >
        <div
          class="app-dual-ring-spinner"
        />
      </div>
      <div
        class="container"
      >
        <div
          class="filters-container"
          data-testid="filters-container"
        >
          <ul
            class="app-tabs"
          >
            <li
              class="app-tab active"
            >
              all
            </li>
            <li
              class="app-tab"
            >
              comedy
            </li>
            <li
              class="app-tab"
            >
              crime
            </li>
            <li
              class="app-tab"
            >
              documentary
            </li>
            <li
              class="app-tab"
            >
              horror
            </li>
          </ul>
          <div
            class="dropdown-container"
          >
            <span
              class="label"
            >
              SORT BY
            </span>
            <div
              class="dropdown-menu"
            >
              <a
                class="app-button"
                data-testid="filters-dropdown-button"
              >
                release date
              </a>
              <ul
                class="app-dropdown"
                data-testid="filters-dropdown"
              >
                <li
                  class="app-dropdown-item"
                  data-testid="dropdown-item"
                >
                  rating
                </li>
                <li
                  class="app-dropdown-item"
                  data-testid="dropdown-item"
                >
                  release date
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="no-results-container"
        >
          Sorry, but nothing matched your search criteria.
        </div>
      </div>
    </main>
  </div>
  <footer
    class="app-footer"
  >
    MovieSearchApp
  </footer>
</DocumentFragment>
`;
